manifests/config.pp

```diff
@@ -40,19 +40,30 @@
     require => File[$gitlab_config_dir],
   }
     # Some versions of gitlab require nginx to be stopped before upgrading: https://github.com/gitlabhq/gitlabhq/issues/7902
-  exec { 'stop gitlab':
-    refreshonly => true,
-    timeout     => 1800,
-    command     => '/usr/bin/gitlab-ctl stop nginx',
-    subscribe   => Package['gitlab'],
-    notify      => Exec['/usr/bin/gitlab-ctl reconfigure'],
-    before      => [ Exec['/usr/bin/gitlab-ctl reconfigure'], Exec['start gitlab'] ],
+  if $::gitlab::gitlab_branch >= '7.10.0' {
+    exec { 'stop gitlab':
+      refreshonly => true,
+      timeout     => 1800,
+      command     => '/usr/bin/gitlab-ctl stop nginx',
```

Comment:
This was Spencer's original code, and is required since the omnibus gitlab package is managed by gitlab-ctl commands.